docs/signing_notes_dogecoin_core_1_21_experimental.md

```diff
@@ -0,0 +1,165 @@
+# Signing Notes
+
+Writing up my notes on the transaction creation and signing process so they're
+readily available to refer to later, and because they may be useful to others.
+
+All of this is tested on an experimental branch of Dogecoin 1.21; the theory is
+generally applicable but other tools will have different interfaces.
+
+## Pre-requisites
+
+In order to sign the output transaction, the PSBT needs to include the details
+of the input transaction(s). As nodes do not track transactions unless they're
+relevant to an address the node is watching, the address the transaction is to
+needs to be watched.
+
+This only needs to be done once, as the transaction creator can add these to
+the PSBT and then circulate the expanded PSBT.
+
+If this is done before the address receives any funds, you can avoid rescanning,
+but as the address now exists the node needs to rescan the blockchain, which
+takes around an hour.
+
+I ran the following to add the address:
+
+```shell
+$ dogecoin-cli importaddress 9xEP9voiNLw7Y7DS87M8QRqKM43r6r5KM5 "corefunds" true
```

Comment:
I think you can add `4532328` as the final param to start the rescan from the first block containing a transaction to this address.